Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -185,26 +189,35 @@ public struct TargetSourcesBuilder {
     }
 
     /// Compute the rule for the given path.
-    private func computeRule(for path: AbsolutePath) -> FileRuleDescription.Rule {
+    private static func computeRule(for path: AbsolutePath,
+                                    toolsVersion: ToolsVersion,
+                                    additionalFileRules: [FileRuleDescription],
+                                    observabilityScope: ObservabilityScope) -> FileRuleDescription.Rule {
+        let rules = Self.rules(additionalFileRules: additionalFileRules, toolsVersion: toolsVersion)
+        // For now, we are not passing in any declared resources or sources here and instead handle any generated files automatically at the callsite. Eventually, we will want the ability to declare opinions for generated files in the manifest as well.
+        return Self.computeRule(for: path, toolsVersion: toolsVersion, rules: rules, declaredResources: [], declaredSources: nil, observabilityScope: observabilityScope)
```

Comment:
intentional that declaredResources is empty array and declaredSources is nil? should they be consistent?

Reply:
Yah, it's intentional since they get different treatment in the existing code.